### PR TITLE
Apple rosetta

### PR DESCRIPTION
**Pull request by kristian240**
### Description

As agreed in [Discord](https://discord.com/channels/686873244791210014/1130493830647533568/1134961449614184590) I added a note in docs about MacOS Rosetta.

The next step is to detect the Apple SIlicon, check if the Rosetta is installed, and prompt the user to install it if needed. 

```shell
# check if apple silicon
uname -m # returns arm64 if Apple Silicon

# check if Rosetta is installed
pgrep -q oahd # returns a number if is installed
```

### Select what type of change this PR introduces:

1. [x] **Just code/docs improvement** (no functional change). 
2. [ ] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### Update Waspc ChangeLog and version if needed
If you did a bug fix, new feature, or breaking change, that affects waspc, make sure you satisfy the following:
1. [ ] I updated [ChangeLog.md](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
2. [ ] I bumped waspc version in [waspc.cabal](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.

### CODE REVIEW

**Comment by matijaSos** [comment]
Awesome work @kristian240! Could you just please rebase this to `release` branch? That's where we keep the docs, while `main` is for the latest product development.

**Comment by kristian240** [comment]
@matijaSos all done!

**Comment by kristian240** [comment]
Ping @matijaSos

**Comment by Martinsos** [comment]
I come a bit late to the party, but we already had a PR for Rosetta, I just remembered, it however never got merged although it was almost there: https://github.com/wasp-lang/wasp/pull/1431 .

Should we combine these two and get the best out of both? Anything to salvage from that one @kristian240 ?

**Comment by kristian240** [comment]
If my interpretation is accurate, the two PRs have different domains. The one from @Zeko369 has more in-depth explanations while this PR makes sure that the majority do not need to go to Discord to find out that Rosetta is needed. Does this make sense?

**Review by Martinsos — COMMENTED** [review]
Apologies for just throwing in that other PR, I was a bit on the go and didn't properly study it. What you said makes sense!

So this is looking good and is a welcome addition, I just had a couple of suggestions to consider before we merge:

Let's put this into
```
:::note Something Unclear?
TEXT HERE
:::
```
to have it be in a nice "NOTE" box!

Also, some ideas for the wording:
1. Add `(arm64)` after Apple Silicon? Or not?
2. I would avoid "you might" here and reword it to "Encountering 'Bad CPU type in executable' on a device ... ?".
3. I would maybe explain a bit why installing Rosetta is an issue -> That MacOS wasp binary is not currently pre-built for arm64 (Apple Silicon) but for x86 and that to get it running they can install Rosetta, which runs x86 stuff on arm64.

These are just suggestions, I am 100% sure in them, so do what you think makes sense and pls argument why not if it doesn't make sense.

**Comment by Martinsos** [comment]
Oh and checking if Rosetta is installed @kristian240  -> that is a cool idea!
Here is our installer if you want to give it a go: https://github.com/wasp-lang/get-wasp .

**Comment by kristian240** [comment]
@Martinsos i changed the copy and place this in a NOTE box. Checkout the ss below.

image

**Comment by kristian240** [comment]
@matijaSos Take a look at https://github.com/wasp-lang/get-wasp/pull/7

File: web/docs/introduction/getting-started.md [hunk 1]
```diff
@@ -107,12 +107,11 @@ Open your terminal and run:
 curl -sSL https://get.wasp-lang.dev/installer.sh | sh
 ```
 
-If you are using macOS on a device with Apple Silicon you might encounter `Bad CPU type in executable` issue. To fix that, please [install Rosetta on your Mac](https://support.apple.com/en-us/HT211861).
-
-```shell
-softwareupdate --install-rosetta
-```
+:::note Something Unclear?
```
Comment (Martinsos) [inline web/docs/introduction/getting-started.md:110]: I merely copied the note block from above in the file with "Something unclear?", I didn't mean this title to be kept for here, my bad I should have made that clear!
Here is suggested change from my side:

```suggestion
:::note Running Wasp on Mac with Mx chip (arm64)
```

File: web/docs/introduction/getting-started.md [hunk 2]
```diff
@@ -107,12 +107,11 @@ Open your terminal and run:
...
+**Experiencing the 'Bad CPU type in executable' issue on a device with arm64 (Apple Silicon)?**
 
+Given that the wasp binary is built for x86 and not for arm64 (Apple Silicon), you'll need to install [Rosetta on your Mac](https://support.apple.com/en-us/HT211861). Rosetta is a translation process that enables users to run applications designed for x86 on arm64 (Apple Silicon). To install Rosetta, run the following command in your terminal: `softwareupdate --install-rosetta`. Once Rosetta is installed, you should be able to run Wasp without any issues.
```
Comment (Martinsos) [inline web/docs/introduction/getting-started.md]:
```suggestion
Given that the wasp binary is built for x86 and not for arm64 (Apple Silicon), you'll need to install [Rosetta on your Mac](https://support.apple.com/en-us/HT211861) if you are using a Mac with Mx (M1, M2, ...) chip. Rosetta is a translation process that enables users to run applications designed for x86 on arm64 (Apple Silicon). To install Rosetta, run the following command in your terminal: `softwareupdate --install-rosetta`. Once Rosetta is installed, you should be able to run Wasp without any issues!
```

**Review by Martinsos — COMMENTED** [review]
@kristian240 nice, here is last round of suggestions from my side!

File: web/docs/introduction/getting-started.md (hunk 2 above)
Comment (kristian240) [reply web/docs/introduction/getting-started.md]: done

**Review by Martinsos — APPROVED** [review]
Awesome, that's it! Thanks, merging!

**Comment by matijaSos** [comment]
Congrats @kristian240 on your first commit and becoming a contributor! :) Thanks for your patience.
